### PR TITLE
[MX-444] Remove iOS components

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Enhancement**

This PR resolves **[MX-444]**

### Description
This PR removes all iOS components from palette. 
_See RFC for full context https://github.com/artsy/README/issues/318_

**Steps to remove a component**
1 - Add `Component.shared.tsx` content to `Component.tsx`
2- Remove `Component.shared.tsx`, `Component.ios.tsx`
3- Remove ios tests from `Component.test.tsx`
4- Fix any broken file imports if any

**Components to be adjusted**
- [x] Avatar - 
- [x] Banner - 
- [x] BarChart - 
- [x] BorderBox - 
- [x] Box - 
- [x] Button - 
- [x] CSSGrid - 
- [x] Cards - 
- [x] Checkbox - 
- [x] CleanTag - 
- [x] Clickable - 
- [x] Collapse - 
- [x] Colors - 
- [x] DataVis - 
- [x] Dialog - 
- [x] DonutChart - 
- [x] EntityHeader - 
- [x] Flex - 
- [x] Grid - 
- [x] HTML - 
- [x] Image - 
- [x] Input - 
- [x] Join - 
- [x] LabeledRange - 
- [x] LineChart - 
- [x] Link - 
- [x] Menu - 
- [x] Message - 
- [x] Modal - 
- [x] Pagination - 
- [x] PriceRange - 
- [x] ProgressBar - 
- [x] ProgressBarTimer - 
- [x] Radio - 
- [x] RadioGroup - 
- [x] ReadMore - 
- [x] ResponsiveBox - 
- [x] Select - 
- [x] Separator - 
- [x] Skeleton - 
- [x] Slider - 
- [x] Spacer - 
- [x] Spinner - 
- [x] StackableBorderBox - 
- [x] StaticCountdownTimer - 
- [x] Stepper - 
- [x] Tabs - 
- [x] Tags - 
- [x] Text - 
- [x] TextArea - 
- [x] TimeRemaining - 
- [x] Toggle - 
- [x] Tooltip - 
- [x] Touchable - 
- [x] Typography - 
- [x] platform -
- [x] svgs - 
- [x] primitives - 

**Final step**
- [x] Adjust documentation

## Follow-up
See https://artsyproduct.atlassian.net/browse/MX-445

[MX-444]: https://artsyproduct.atlassian.net/browse/MX-444

### CODE REVIEW

File: packages/palette/package.json [hunk 1]
```diff
@@ -22,8 +22,7 @@
     "release": "auto shipit && node scripts/clean-up-after-auto.js",
     "test": "yarn type-check && yarn jest --runInBand",
     "test:watch": "jest --watch --runInBand",
-    "type-check": "tsc --noEmit --pretty && yarn type-check-ios && yarn type-check-scripts",
-    "type-check-ios": "tsc -p tsconfig.ios.json --noEmit --pretty",
+    "type-check": "tsc --noEmit --pretty && yarn type-check-scripts",
     "type-check-scripts": "tsc -p tsconfig.scripts.json --noEmit --pretty",
     "type-declarations": "tsc --emitDeclarationOnly",
     "watch": "concurrently --raw --kill-others 'yarn compile -w' 'yarn type-declarations -w'",
```
Comment (ds300) [inline packages/palette/package.json:28]: you should also delete the line below this (github won't let me add a comment there 🙈 )

Comment (ds300) [reply packages/palette/package.json:28]: there's also the `"clean:eigen"` and `"compile:eigen"` commands in this file

**Review by ds300 — APPROVED** [review]
This is great work, thanks! 🙇 

There's a few more commands to delete in `packages/palette/package.json` but feel free to merge after  🍏

File: packages/palette/package.json (hunk 1 above)
Comment (MounirDhahri) [reply packages/palette/package.json:25]: sure 🎉

**Comment by artsyit** [comment]
:rocket: PR was released in [`v13.0.0`](https://github.com/artsy/palette/releases/tag/v13.0.0) :rocket:

**Comment by damassi** [comment]
Thank you @MounirDhahri 👍

**Pull request merged**